### PR TITLE
Move ValueList to functions/lib/aggregates for Spark reuse

### PR DESCRIPTION
**Pull request by liujiayi771**
The array_agg function in Spark and Presto has some inconsistent behavior
when handling null values. We need to re-implement Spark's array_agg in Velox.
To be able to reuse `ValueList`, it needs to be moved to the
`functions/lib/aggregates` directory first.

The `copyValueListToArrayWriter` method has also been moved to the
`ValueList`, making it reusable.

### CODE REVIEW

**Comment by mbasmanova** [comment]
> The array_agg function in Spark and Presto has some inconsistent behavior when handling null values. We need to re-implement Spark's array_agg in Velox.

What is the difference in behavior between Presto and Spark?

File: velox/functions/lib/aggregates/ValueList.h [hunk 1]
```diff
@@ -19,9 +19,12 @@
 #include "velox/common/memory/HashStringAllocator.h"
 #include "velox/exec/Aggregate.h"
 #include "velox/expression/ComplexViewTypes.h"
+#include "velox/expression/ComplexWriterTypes.h"
 #include "velox/vector/ComplexVector.h"
 #include "velox/vector/DecodedVector.h"
 
+using namespace facebook::velox::exec;
```
Comment (mbasmanova) [inline velox/functions/lib/aggregates/ValueList.h:26]: no using namespace in headers, please

**Comment by liujiayi771** [comment]
> What is the difference in behavior between Presto and Spark?

The semantics are generally consistent, but there are inconsistencies between Presto and Spark in the handling of null values. Spark always ignores null values in the input, whereas Presto preserves them. Moreover, Presto returns null when all inputs are null, while Spark returns an empty list or set.
I plan to rewrite Spark's array_agg based on the simple aggregate function interface.

**Comment by liujiayi771** [comment]
Hi @mbasmanova, there is a draft PR https://github.com/facebookincubator/velox/pull/9231 for Spark's `array_agg`, you could get more details from it. I plan to split the move of `ValueList` and the implementation of Spark's `array_agg` into two PRs, although they could be combined into a single PR.

**Comment by mbasmanova** [comment]
@liujiayi771 Thank you for clarifying. Presto's implementation depends on presto.array_agg.ignore_nulls config flag. Does Spark implementation match Presto's when presto.array_agg.ignore_nulls = true?

**Comment by mbasmanova** [comment]
@liujiayi771 

> When all inputs are null, the output of Spark is an empty array, while Presto returns null.

Got it. Thank you for clarifying.

CI is red. Please, take a look.

**Comment by mbasmanova** [comment]
@liujiayi771 It might be #9200

**Review by mbasmanova — COMMENTED** [review]
@kagamiori Wei, would you help review this PR?

**Review by pedroerp — APPROVED** [review]
Looks good to me. Thanks @liujiayi771

**Review by Yuhta — APPROVED** [review]

**Comment by liujiayi771** [comment]
@Yuhta @mbasmanova Did we encounter any issues with the merge? Do I need to rebase with the latest code?

**Comment by pedroerp** [comment]
> @Yuhta @mbasmanova Did you encounter any issues with the merge? Do I need to rebase with the latest code?

There were a few smaller internal fixes required since the header got moved, but I'm merging it today.

**Pull request closed without merging**